### PR TITLE
Update i18n docs to use make-json command from wp-cli

### LINKED ISSUE #14093: Update Internationalization with wp-cli make-json
This issue is for keeping track the internationalization documentation needs to be updated to use `wp i18n make-json` when this issue is fixed: https://github.com/wp-cli/i18n-command/issues/146

I'm separating out this issue from #13909 so the documentation can be updated and usable in the mean time.

### PR DESCRIPTION
**Pull request by mkaz**
## Description

Updates the Internationalization documentation recommending to use the wp-cli make-json command to generate the necessary json file needed for translations.

This replaces the instructions to use the npm po2json module, it is simpler for a user to use only the wp-cli tool which is also used earlier in the documentation for creating the pot files.

There was a bug fixed in wp-cli https://github.com/wp-cli/i18n-command/issues/146

Fixes #14093

## How has this been tested?

Follow instructions in documentation to confirm JSON file is created and translations work.

## Types of changes

Documentation.

### CODE REVIEW

**Comment by dianeco** [comment]
@mkaz I've followed your instructions. I have a plugin that create custom blocks which is divided into 11 .js files that contain translation strings. For the last step, when I use wp-cli to convert the po file to JSON, it creates 11 plugin-eo-[md5].json files. Is there an extra step to merge all these files into one?

**Comment by mkaz** [comment]
@dianeco The language files are directly related to the JavaScript files, so when a JS file is enqueued WP will use the related language files when needed. So you can combine the JS together prior to translations, or it is fine to have all the additional language files.

**Comment by dianeco** [comment]
@mkaz Thanks for your answer. I use [create-guten-block](https://github.com/ahmadawais/create-guten-block) which combine the files into one but the command `wp i18n make-pot ./ languages/myguten.pot` doesn't succeed to add the translatable javascript strings into the pot file. Any idea?

**Comment by swissspidy** [comment]
@dianeco Make sure you use the correct text domain throughout all of your PHP and JavaScript code. This is a best practice and required for string extraction to work properly across the board. `wp i18n make-pot` skips any strings with wrong missing text domains unless you use `--ignore-domain`.

**Comment by dianeco** [comment]
@swissspidy Thanks for your reply. Yes I use the correct text domain in the JavaScript code. The process described above works fine when I use it on the dev files (but it creates a json file for each js file). 
However on production mode with all my js files combined into one, `wp i18n make-pot` doesn't extract the translatable javascript strings even if I use `--ignore-domain`.

**Comment by swissspidy** [comment]
That sounds like your production JS files is all minified, so instead of having `wp.i18n.__()` in it there will be something like `xyz()`. But WP-CLI can only find `__()` calls. I suggest you to report an issue with the create-guten-block repository.

**Comment by mkaz** [comment]
@swissspidy Can you take a look at this PR? I'd love to get the instructions to use make-json instead of po2json updated and out to users. Thanks!

File: docs/designers-developers/developers/internationalization.md [hunk 1]
```diff
@@ -176,31 +176,32 @@ msgid "Hello World"
 msgstr "Saltuon mundo"
 ```
 
-The last step to create the translation file is to convert the `myguten-eo.po` to the JSON format needed. For this, you can use the [po2json utility](https://github.com/mikeedwards/po2json) which you install using npm. It might be easiest to install globally using: `npm install -g po2json`. Once installed, use the following command to convert to JED format:
+The last step to create the translation file is to convert the `myguten-eo.po` to the JSON format needed. For this, you can use the wp-cli [make-json command](https://developer.wordpress.org/cli/commands/i18n/make-json/), requires wp-cli v2.2.0 and later.
```
Comment (swissspidy) [inline docs/designers-developers/developers/internationalization.md]:
```suggestion
The last step to create the translation file is to convert the `myguten-eo.po` to the JSON format needed. For this, you can use WP-CLI's [`wp i18n make-json` command](https://developer.wordpress.org/cli/commands/i18n/make-json/), which requires WP-CLI v2.2.0 and later.
```

File: docs/designers-developers/developers/internationalization.md [hunk 2]
```diff
@@ -222,7 +223,7 @@ The final part is to tell WordPress where it can look to find the translation fi
 
 WordPress will check for a file in that path with the format `${domain}-${locale}-${handle}.json` as the source of translations. Alternatively, instead of the registered handle you can use the md5 hash of the relative path of the file, `${domain}-${locale} in the form of ${domain}-${locale}-${md5}.json.`
 
-This example uses the handle, rename the `myguten-eo.json` file to `myguten-eo-myguten-script.json`.
+Using `make-json` automatically names the file with the md5 hash, so it is ready as-is. You could rename to use the handle, in this case would be `myguten-eo-myguten-script.json`.
```
Comment (swissspidy) [inline docs/designers-developers/developers/internationalization.md]:
```suggestion
Using `make-json` automatically names the file with the md5 hash, so it is ready as-is. You could rename the file to use the handle instead, in which case the file name would be `myguten-eo-myguten-script.json`.
```

**Review by swissspidy — APPROVED** [review]

**Pull request merged**